docs: webhook secret error in the PR (#213)
konflux-ci/konflux-ci: docs/troubleshooting.md

6. :gear: On the PR page, type `/retest` on the comment box and post the comment.
   Observe the behavior once more.

# PR Fails when Webhook Secret was not Added

> When creating a Github Apps if webhook secret was not added we get following error in

Reviewer: ```suggestion
When creating a Github App, if webhook secret was not added we get following error in
```

> When creating a Github Apps if webhook secret was not added we get following error in
> our opened PR pipeline logs:

Reviewer: ```suggestion
If a webhook secret is not added when creating a Github App,  PR pipelines will fail and the following error will show up in the PR logs:
```

```
There was an issue validating the commit: "could not validate payload, check your webhook secret?: no signature has been detected, for security reason we are not allowing webhooks that has no secret"
```

> To resolve this issue, go to the Github Apps you created and check Webhook Secret is

Reviewer: ```suggestion
To resolve this issue, go to the Github App you created and check Webhook Secret is
```

> added there. Please refer 

Reviewer: ```suggestion
added there. Please refer to
```

[this document](https://pipelinesascode.com/docs/install/github_apps/#manual-setup)

> on how to create Webhook Secret.

Reviewer: To be consistent with the rest of the document:
```suggestion
on how to create webhook secret.
```

6. :gear: On the PR page, type `/retest` on the comment box and post the comment.
   Observe the behavior once more.

> # PR fails when Webhook Secret was not added

Reviewer: Match capitalization to other titles on the doc.
```suggestion
# PR Fails when Webhook Secret was not Added
```

> When creating a github application if webhook secret was not added we get following error in our opened PR:

Reviewer 2: ```suggestion
When creating a github application if webhook secret was not added we get the following error in our opened PR:
```

> When creating a github application if webhook secret was not added we get following error in our opened PR:

Reviewer: Limit lines with simple text to 89 characters.

> When creating a github application if webhook secret was not added we get following error in our opened PR:

Reviewer: github -> GitHub

> When creating a github application if webhook secret was not added we get following error in our opened PR:

Reviewer: application -> Apps
In github docs, they are referred to as GitHub Apps

> When creating a github application if webhook secret was not added we get following error in our opened PR:

Reviewer: ```quote
 we get following error in our opened PR
```
Let's be more specific about where we see it. In the Konflux pipeline logs? In the status that is reported to the PR?

```
There was an issue validating the commit: "could not validate payload, check your webhook secret?: no signature has been detected, for security reason we are not allowing webhooks that has no secret"
```

To resolve this issue, go to the github application you created and add your webhook secret which was generated using following command

> ```bash

Reviewer 2: I am not sure about this command
I think its enough to state that we need to add the secret that we got when we created the application (or generate a new one if we do not know what it is)

Reviewer: I agree. I think we should point the reader to the relevant part of the documentation rather than repeating it, and ask them to confirm that the app includes a secret.

Specifically here, this command is not a part of our documentation - it's mentioned in pipelines as code documentation, so if it changes, we'd have to change our documentation as well.
